### PR TITLE
Microverse - Newsweek clone using bootstrap

### PR DESCRIPTION
**Pull request by Maclenn77**
In this project, I created a clone of Newsweek using bootstrap. This website has:
-   One navbar with a toggle menu for sm and md sizes.
-   A main section with many different articles. Each article rearrange in each breakpoint
-   A footer with other info from the website.

I used media queries and bootstrap for a responsive design with two breakpoints. Thank you for reviewing!

### CODE REVIEW

**Comment by uryelah** [comment]
Hello ** J. P. Pérez**, I'm Sarah, your reviewer for this PR 👩‍🚀  

**First of all good job at for a first PR.🎇**
Great use of semantic HTML and attention to colors!

There are still some issues to work through so hang in there 💪 

---------    

### Required changes 🛠 
---------

- [x] Since a lot of the responsiveness and style are being handled by Bootstrap this project needs to be as close as possible to the original site. This includes **spacing**, **positioning** and **font style and weight**. Compare your clone and [the original site ](https://www.newsweek.com/)
carefully and find the differences in style so they can get fixed.

- [x] The original site doesn't take 100% of the width after a certain window size. Compare you version and the original site bellow:
Screenshot_2020-03-20 Newsweek clone

**Original:**
Screenshot_2020-03-20 Newsweek(13)
-------

- [x] Your whole layout, but the navbar and footer, is breaking at screens smaller or equal to 991px of width. Try to fix that using mainly Boostrap responsive grid system. [Check this article about it here.](https://uxplanet.org/how-the-bootstrap-4-grid-works-a1b04703a3b7)
Screenshot_2020-03-20 Newsweek clone(1)

-------

- [x] There's a slight horizontal scroll in your whole site's body in wide screens. Please fix that overflow issue.

- [x] Try to style the navbar search bar in a similar fashion to the original one.
Screenshot_2020-03-20 Newsweek clone(2)
**Original search:**
Screenshot_2020-03-20 Newsweek(14)

--------

- [x] Make sure that the 'Featured Stories', 'Top story', and 'Opinion' sections have margins between them. Look for a way to add margins with Bootstrap classes.

- [x] At the 'Opinion' section add pictures of the colonists that are equal from the ones in the original site.
**Your pictures**
Screenshot_2020-03-20 Newsweek clone(3)

**Original ones**
Screenshot_2020-03-20 Newsweek(15)

--------

- [x] The 'Newsletter' image should remain proportional and not distorted regardless of the screen size. Fix it so it's not distorted anymore.
Screenshot_2020-03-20 Newsweek clone(4)

**Original one:**
Screenshot_2020-03-20 Newsweek(16)

----------

## Optional Improvements

- [x] [OPTIONAL] Put a screenshot of your clone in your readme file to show off your work

--------

I share the sentiment that _Bootstrap is ok_ and I know that parsing through the documentation can be frustrating at times, I really like [cheat-sheet](https://hackerthemes.com/bootstrap-cheatsheet/) for my own workflow, but if you give yourself some time to play around with the classes and test how they work I think you'll be pleasantly surprised about how bootstrap can make responsiveness practical.
This project is on the bigger side but I'm sure you'll be able to nail it!

**Kindly fix the REQUIRED issues and send another code review request.**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by Maclenn77** [comment]
Thanks a lot, @uryelah ! It has been the best code review that I've received.
The Cheatsheet and the link that you shared were really useful.
I've done the changes and I hope there's any missing.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,40 @@
-# Newsweek_clone_bootstrap
-Microverse Project for learning to use bootstrap. This is a clone of Newsweek clone using this framework.
+## Building with Bootstrap - Newsweek
+
+>This project consists of replicating the <a href="https://www.newsweek.com/" target="_blank">Newsweek</a>💻 website.
+Newsweek is one of many site that uses Bootstrap for its design.
```
Comment (majovanilla) [inline README.md:4]: - [ ] Improve your Readme file. It should be descriptive of the project. You are using the correct template but now you need to fill it with information about your project. You can add details about what you did in this project, why you did it, how you did it...Check this [link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/) for more info.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,874 @@
...
+          <time class="d-none d-lg-block" datetime="2020-03-20">Fri, Mar 20, 2020</time>
+        </div>
+        <div class="align-self-end d-block m-3 order-lg-3 h5">
+          <span class="text-uppercase font-weight-bold text-white px-2"><a href="#">Sign in</a></span>
+          <span class="text-uppercase font-weight-bold bg-white px-2 py-1"><a href="#" id="sign">Subscribe ></a></span>
+        </div>
+        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+        </button>
+      </div>
```
Comment (majovanilla) [inline index.html]: - [ ] Your navbar has more height than the original one. Please make the adjustment.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,874 @@
...
+              <li class="nav-item">
+                <h6><a class="nav-link text-nowrap" href="#">The Debate <span class="large text-muted"> | </span><span class="short"> > </span></a></h6>
+              </li>
+              <li class="nav-item">
+                <h6><a class="nav-link text-nowrap" href="#">Vantage <span class="large text-muted"> | </span><span class="short"> > </span></a></h6>
+              </li>
+              <li class="nav-item">
+                <h6><a class="nav-link text-nowrap" href="#">Weather <span class="short"> > </span></a></h6>
+              </li>
+        </ul>
```
Comment (majovanilla) [inline index.html]: - [ ] The navbar menus are not displaying like the original page. You should check the width of the `<ul>` element and make it look like the original page.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,874 @@
...
+        </form>
+      </div>
+    </nav>
+    <!-- Main section  -->
+  <main class="grid-area bg-white">
+      <!-- Top story -->
+      <article class="top-story mb-4 bg-white">
+        <h5 class="text-muted text-uppercase">Top Story</h5>
+        <img class="img-fluid" src="img/bootstrap-top.jpg" alt="Bootstrap">
+        <h2 class="text-weight-bold">Bootstrap makes building a site faster, developers say</h2>
```
Comment (majovanilla) [inline index.html]: - [ ] The elements inside your `<main>` tag should have more margin to the sides. Please use bootstrap classes for this.

File: index.html [hunk 5]
```diff
@@ -0,0 +1,874 @@
...
+        <div class="d-flex justify-content-center">
+          <span class="features h4 text-white pr-1 pl-1 font-weight-bold mr-2">N</span> <h2 class="text-center font-weight-bold h4 m-2">Editor's pick</h2>
+        </div>
+        <hr>
+        <div class="row row-cols-md-2">
+          <section class="travel-container col-md">
+            <figure class="img-container m-auto">
+              <img src="img/yellowstone.jpg" alt="Yellowstone" class="img-fluid">
+              <figcaption class="featured-image pl-2 pr-2 font-weight-bold">
+                Tech & Science
```
Comment (majovanilla) [inline index.html]: - [ ] The badges you are using inside the images of this section should be located on the bottom-left corner of the image. Try using `position: absolute` to keep them inside the image. Check this out in the Slideshows section, too.

File: index.html [hunk 6]
```diff
@@ -0,0 +1,874 @@
...
+              <h3 class="font-weight-bold">Winter is coming!</h3>
+              <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
+            </div>
+          </article>
+      </section>
+    <!-- More stories -->
+      <article class="more-stories bg-white">
+        <h5 class="text-muted text-uppercase">More stories</h5>
+        <section class="border-top">
+          <img src="img/bootstrap.png" alt="Image Debate" class="float-left mr-4">
```
Comment (majovanilla) [inline index.html]: - [ ] The articles in this section have a larger image when you are in the medium breakpoint and below. Please adjust them.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,874 @@
...
+        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. </p>
+
+        <img class="img-fluid feat-extra" src="img/image-2.png" alt="Image 2">
+        <h2 class="text-weight-bold feat-extra"><a href="#">Is Bootstrap 4 the future framework?</a></h2>
+        <p class="feat-extra">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. </p>
+      </article>
+      <!-- Opinion section-->
+      <article class="opinion-section bg-white">
+        <h5 class="text-muted text-uppercase">Opinion</h5>
+        <section class="d-flex">
```
Comment (majovanilla) [inline index.html]: - [ ]  The images in this section are larger in the medium and small breakpoints. Also, check the font-size and adjust it.

File: index.html [hunk 8]
```diff
@@ -0,0 +1,874 @@
...
+        </div>
+      </article>
+    <!-- Latest News section -->
+      <article class="latest-news bg-white">
+        <h5 class="text-muted text-uppercase">Latest News</h5>
+          <section class="d-flex">
+            <img src="img/bootstrap.png" alt="Image Debate" class="float-left mr-4">
+            <h6 class="font-weight-bold clearfix"><a href="#">CSS is easier with bootstrap!</a></h6>
+          </section>
+          <section class="border-top d-flex">
```
Comment (majovanilla) [inline index.html]: - [ ] The articles in this section have a larger image when you are in the medium breakpoint and below. Please adjust them.

File: index.html [hunk 9]
```diff
@@ -0,0 +1,874 @@
...
+        <h6 class="f-category">Newsletter</h6>
+        <figure class="text-center border-muted border m-auto d-md-flex">
+          <img src="img/newsletter.png" alt="Newsletter" class="img-sm-fluid img-md-fluid img-xm-fluid">
+          <figcaption class="text-center border-muted p-3 d-md-flex flex-md-column align-items-start">
+            <h5 class="font-weight-bold">The Starting 5</h5>
+            <p class="text-md-left">Take back your inbox. Sign up for our newsletter and start making sense of the world around you.</p>
+            <a href="#" class="text-uppercase text-center p-2 anchor-news">Sign Up</a>
+          </figcaption>
+        </figure>
+      </article>
```
Comment (majovanilla) [inline index.html]: - [x] The elements in this section are not behaving as expected. Check the image below and see how the newsletter is overflowing the body of your page. If you make use of the suggested classes in the body, you might be able to solve this problem.

- [x] You are not using the `col-*` class in these elements. You should take advantage of the bootstrap classes to achieve the desired design.

Screenshot from 2020-03-23 18-31-34

Original:
Screenshot from 2020-03-23 18-33-29

File: index.html [hunk 10]
```diff
@@ -0,0 +1,874 @@
...
+        <li class="list-inline-item"><a href="#">About us</a><span class="text-secondary d-sm-none d-md-inline"> |</span></li>
+        <li class="list-inline-item"><a href="#">About us</a><span class="text-secondary d-sm-none d-md-inline"> |</span></li>
+        <li class="list-inline-item"><a href="#">About us</a><span class="text-secondary d-sm-none d-md-inline"> |</span></li>
+        <li class="list-inline-item"><a href="#">About us</a><span class="text-secondary d-sm-none d-md-inline"> |</span></li>
+        <li class="list-inline-item"><a href="#">About us</a><span class="text-secondary d-sm-none d-md-inline"> |</span></li>
+        <li class="list-inline-item"><a href="#">About us</a><span class="text-secondary d-sm-none d-md-inline"> |</span></li>
+        <li class="list-inline-item"><a href="#">About us</a><span class="text-secondary d-sm-none d-md-inline"> |</span></li>
+      </ul>
+    </section>
+  </footer>
```
Comment (majovanilla) [inline index.html]: - [ ] The footer has a 100% width and it should have the same width as the rest of the page (not including the navbar). Please make the adjustment.

File: styles/style.css [hunk 11]
```diff
@@ -0,0 +1,459 @@
...
+
+/* Marked important to overlap bootstrap rules */
+
+h6 {
+  color: #000 !important;
+}
+
+nav {
+  padding: 0 !important;
+}
```
Comment (majovanilla) [inline styles/style.css]: - [ ] You should use bootstrap classes.

File: styles/style.css [hunk 12]
```diff
@@ -0,0 +1,459 @@
...
+  width: 3.5rem;
+}
+
+img[alt="Image Debate"] {
+  height: 5rem;
+}
+
+img[alt="weather"] {
+  width: 1.2rem;
+}
```
Comment (majovanilla) [inline styles/style.css]: - [ ] - [ ] You should use bootstrap classes.

File: styles/style.css [hunk 13]
```diff
@@ -0,0 +1,459 @@
...
+
+.country {
+  display: none;
+}
+
+footer {
+  background-color: #fff;
+  padding: 1rem 0.8rem;
+  margin-top: 2rem;
```
Comment (majovanilla) [inline styles/style.css]: - [ ] You should use bootstrap classes.

File: styles/style.css [hunk 14]
```diff
@@ -0,0 +1,459 @@
...
+}
+
+h4,
+h3 {
+  color: #000;
+}
+
+h5 {
+  font-size: 16px;
+  font-weight: bold !important;
```
Comment (majovanilla) [inline styles/style.css]: - [ ] You should use bootstrap classes.

File: styles/style.css [hunk 15]
```diff
@@ -0,0 +1,459 @@
...
+  color: #fff !important;
+}
+
+a:hover[class="short"] {
+  color: red !important;
+}
+
+/* Featured posts */
+.featured-image {
+  position: absolute;
```
Comment (majovanilla) [inline styles/style.css]: - [ ] You should use bootstrap classes.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hey @Maclenn77 :wave: 

## Good job implementing the required changes on this project! 👍

There are still some changes 👇 you need to implement to approve this request 🔍

### Required Changes: ✍️

- [x] The previous TSE required you to check the general width of your body elements to match the original page. You used grid to achieve a similar look, but you should take a look at the `container-fluid` bootstrap class and use `col-*` in a responsive way to control the width of the body in the different breakpoint.

- [x] You are not using bootstrap classes **everywhere**. This is a [requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/using_bootstrap/bootstrap_classes_requirement.png) 

- [x] Properties like width, height, padding, and margin should be all over your HTML file and not in your CSS. You **are not using enough col-* classes** and this is the best feature bootstrap has!

- [x] I commented **some** of your CSS classes so that you change them for bootstrap classes. There is a lot more to change but I didn't mention them to avoid repetition. Focus on the properties mentioned above.

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Comment by uryelah** [comment]
Hi **J. P. Pérez**, I'm Sarah, back to review this PR 👩‍🚀

You did a great deal of progress since the first review! I'm really impressed! 🎆 

There are still some style and spacing issues left since, as mentioned before, this project requires a lot of attention to detail. You already did the hardest part though so let's check what's next 💪 

------

## Required changes 🛠 
---------

- [ ] All the sections above 'In the Magazine' need to be aligned at the bottom without any extra empty space. To reach that you can try using the 'overflow' property and set it to scroll vertically. Try to figure out how to hide the scrollbar though.
Screenshot_2020-03-25 Newsweek(31)
**Original site:**
Screenshot_2020-03-25 Newsweek clone(2)

-----

- [x] Good job changing the color of the menu icon, it's really hard to see it right now so try to make it a bit more contrasting. Compare your version and the original one below:

Screenshot_2020-03-25 Newsweek clone(3)
Screenshot_2020-03-25 Newsweek(32)

----

- [x] On medium to small screens the page content takes 100% of the window width:
**Your version still has margins:**
Screenshot_2020-03-25 Newsweek clone(4)
**Original version:**
Screenshot_2020-03-25 Newsweek(33)

-----

- [x] In medium and small-sized screens some of your images, like the ones from the 'Opinion' section are too big. Check the original site again and make sure they are closer to the original proportion displayed in the specific screen sizes.

Screenshot_2020-03-25 Newsweek clone(5)
Screenshot_2020-03-25 Newsweek(34)

---- 

- [x] Make sure you use the same images, or similar ones, in each section header. Compare bellow the 'N' red squares you added and the original ones:


Screenshot_2020-03-25 Newsweek clone(6)
Screenshot_2020-03-25 Newsweek(35)


Screenshot_2020-03-25 Newsweek clone(7)
Screenshot_2020-03-25 Newsweek(36)

----

- [x] On wide screens the footer must have the same width as the content of the site, not 100% of the window width:
Screenshot_2020-03-25 Newsweek clone
Screenshot_2020-03-25 Newsweek(29)


---

- [x] Make sure the 'Latest News' images don't get distorted on resize and keep their proportions, compare your version and the origina site:
Screenshot_2020-03-25 Newsweek clone(1)
Screenshot_2020-03-25 Newsweek(30)

---------

**Kindly fix the REQUIRED issues and send another code review request.**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by men32z — CHANGES_REQUESTED** [review]
## Good job on this project and making changes! 👍

There are still some changes 👇 you need to implement to approve this request 🔍

* [x] In the previous review one of my colleagues mentioned something about color, you can use a tool to pick colors on your browser, try to search "firefox colorpicker" or "chrome colorpicker" there are good tools, you can also inspect the code of the original page until you found the color amoung the clases.  right now I'm going to tell you specifically the color is #f72210

* [x] you did great with the margins for mobile but now the full width version has the 100% of the with and should have the same limit as all page. 
#### original version
Screenshot_2020-03-26 Newsweek
#### your version 
Screenshot_2020-03-26 Newsweek clone

* [x] good job on this change, just try to make it even better by changin the mobile font size and weight(bold)
Screenshot_2020-03-26 Newsweek clone

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@men32z)
I would be happy to help!

--Luis Preza

**Review by davidmrtz-dev — APPROVED** [review]
### **Status: Approved.** ✅ 
****
### Hi @Maclenn77 . 😄 
****
Awesome job!, You correctly implemented all the specifications required by the previous TSE's and your project looks awesome and shows that you know what are you doing 😎 
Congratulations! 👍 
You can merge this branch and move onto the next project. 🎊 
Happy coding! 😄 💻  
****
**[TSE:  David Elí]** 👨‍💻 
Github: [@davitomix](https://github.com/davitomix)
Twitter: [@davitomix](https://twitter.com/davitomix007)
Linkedin: [David Elí Martínez García](https://www.linkedin.com/in/davideligarcia/)

**Pull request merged**